code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -4245,13 +4247,75 @@ void ServerGameState::AttachToObject(fx::ServerInstanceBase* instance)
 
 	auto sbac = fx::StateBagComponent::Create(fx::StateBagRole::Server);
 	sbac->SetGameInterface(this);
 
 	instance->GetComponent<fx::GameServer>()->GetComponent<fx::HandlerMapComponent>()->Add(HashRageString("msgStateBag"), 
-		{ fx::ThreadIdx::Sync, [this](const fx::ClientSharedPtr& client, net::Buffer& buffer)
+		{ fx::ThreadIdx::Sync, [this, instance](const fx::ClientSharedPtr& client, net::Buffer& buffer)
 	{
+		static fx::RateLimiterStore<uint32_t, false> stateBagRateLimiterStore{ instance->GetComponent<console::Context>().GetRef() };
+			
+		// You can adjust this with `set rateLimiter_stateBag_rate [max rate]`
+		// Example: `set rateLimiter_stateBag_rate 10` would only allow for a max of 10 state bags changes before get hit
+		// by the rate limiter
+		constexpr double kStateBagRateLimit = 50.f;
+		// You can adjust this with `set rateLimiter_stateBag_burst [max burst]`
+		// this allows the rate to temporarily increase without instantly spilling over to the flood limiter.
+		constexpr double kStateBagRateLimitBurst = 100.f;
+		static auto stateBagRateLimiter = stateBagRateLimiterStore.GetRateLimiter("stateBag", fx::RateLimiterDefaults{ kStateBagRateLimit, kStateBagRateLimitBurst });
+			
+		// You can adjust this with `set rateLimiter_stateBag_rate [max rate]`
+		// Example: `set rateLimiter_stateBagFlood_rate 10` would only allow for a max of 10 state bags changes after the
+		// the `stateBag` rate limit gets hit.
+		constexpr double kStateBagRateFloodLimit = 75.f;
+		// You can adjust this with `set rateLimiter_stateBagFlood_burst [max burst]`
+		// this allows the rate to temporarily increase without instantly flagging to kick them
+		constexpr double kStateBagRateLimitFloodBurst = 100.f;
+		static auto stateBagRateFloodLimiter = stateBagRateLimiterStore.GetRateLimiter("stateBagFlood", fx::RateLimiterDefaults{ kStateBagRateFloodLimit, kStateBagRateLimitFloodBurst });
+
+		// You can adjust this with `set rateLimiter_stateBagSize_rate [max state bag size]`
+		// Example: `set rateLimiter_stateBagSize_rate 1024` would reduce the maximum state bag size to 1 kilobyte
+		constexpr double kStateBagSizeLimit = 64 * 1024.0;
+		// You shouldn't adjust this value. This value is hard coded to be 128kb internally and should never be larger than this..
+		constexpr double kStateBagSizeLimitBurst = 128 * 1024.0;
+		static auto stateBagSizeRateLimiter = stateBagRateLimiterStore.GetRateLimiter("stateBagSize", fx::RateLimiterDefaults{ kStateBagSizeLimit, kStateBagSizeLimitBurst });
+
+		const uint32_t netId = client->GetNetId();
+
+		if (!stateBagRateLimiter->Consume(netId))
+		{
+			if (!stateBagRateFloodLimiter->Consume(netId))
+			{
+				if (!g_ignoreStateBagWarnings->GetValue() && !client->IsDropping())
+				{
+					console::Printf("state-bag-warning", "Client %s %i got dropped for sending too many state bag value updates.\n", client->GetName(), netId);
```

Comment:
Do we handle channel names with `-` properly in suppression logic?

Reply:
They seem to work fine.